addons/point_of_sale/models/account_journal.py

```diff
@@ -14,3 +14,10 @@ def _check_type(self):
         methods = self.env['pos.payment.method'].sudo().search([("journal_id", "in", self.ids)])
         if methods:
             raise ValidationError(_("This journal is associated with a payment method. You cannot modify its type"))
+
+    def _get_journal_inbound_outstanding_payment_accounts(self):
+        res = super()._get_journal_inbound_outstanding_payment_accounts()
+        account_ids = set(res.ids)
+        for payment_method in self.pos_payment_method_ids:
```

Comment:
```suggestion
        for payment_method in self.sudo().pos_payment_method_ids:
```
Hi @Arnaud-Sibille. I'm currently debugging and it looks like this solution of yours is the correct approach. Now, to get rid of the access error, we need sudo access to the pos_session to get the payment methods.

But before merging, can you show me in action the issue?